File: app/src/main/java/dev/blank/lawancorona/data/local/Preferences.java
package dev.blank.lawancorona.data.local;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import dev.blank.lawancorona.R;

public class Preferences {
    private static final String KEY_WILAYAH = "wilayah";

    private static SharedPreferences getSharedPreference(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static void setWilayah(Context context, String wilayah) {
        SharedPreferences.Editor editor = getSharedPreference(context).edit();
        editor.putString(KEY_WILAYAH, wilayah);
        editor.apply();
    }

    public static String getWilayah(Context context) {
        return getSharedPreference(context).getString(KEY_WILAYAH, context.getString(R.string.dunia));
    }
}

File: app/src/main/java/dev/blank/lawancorona/data/local/Preferences.kt
package dev.blank.lawancorona.data.local

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

object Preferences {
    private const val KEY_WILAYAH = "wilayah"
    private fun getSharedPreference(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

    fun setWilayah(context: Context, wilayah: String?) {
        val editor = getSharedPreference(context).edit()
        editor.putString(KEY_WILAYAH, wilayah)
        editor.apply()
    }

    fun getWilayah(context: Context): String? {
        return getSharedPreference(context).getString(KEY_WILAYAH, "ALL")
    }
}